tmt/utils.py

```diff
@@ -3323,19 +3323,29 @@ def shell_variables(
 
 
 def duration_to_seconds(duration: str) -> int:
-    """ Convert sleep time format into seconds """
+    """ Convert extended sleep time format into seconds """
     units = {
         's': 1,
         'm': 60,
         'h': 60 * 60,
         'd': 60 * 60 * 24,
         }
-    if re.match(r'^(\d+ *?[smhd]? *)+$', str(duration)) is None:
+    # Couldn't create working validation regexp to accept '2 1m4'
+    # thus fixing the string so \b works well
+    fixed_duration = re.sub(r'([smhd])(\d)', r'\1 \2', str(duration))
+    re_validate = r'^((\*\s*\d+(?!\s*[smhd])\b\s*)|(\d+\s*[smhd]?\b)\s*)+$'
+    re_split = r'(?:(?:(\*)\s*(\d+)(?!\s*[smhd])\b\s*)|(?:(\d+)\s*([smhd]?)\b)\s*)'
```

Comment:
Would you consider verbose regular expressions? I can't tell whether the pattern is correct or not without diving deeply into dissecting it (tests would fail, so it probably is, but even the test might be wrong...). Using a verbose approach might improve readability, one can comment on what each component means, e.g. https://github.com/teemtee/tmt/blob/main/tmt/hardware.py#L116:

```python
CONSTRAINT_VALUE_PATTERN = re.compile(rf"""
    ^                                   # must match the whole string
    (?P<operator>{_OPERATOR_PATTERN})?  # optional operator
    \s*                                 # operator might be separated by white space
    (?P<value>.+?)                      # actual value of the constraint
    \s*                                 # there might be trailing white space
    $                                   # must match the whole string, I said :)
    """, re.VERBOSE)
```

Reply:
changed in dd7a4621